agent/src/main/scala/za/co/absa/atum/agent/dispatcher/Dispatcher.scala

```diff
@@ -16,11 +16,11 @@
 
 package za.co.absa.atum.agent.dispatcher
 
-import za.co.absa.atum.agent.AtumContext
-import za.co.absa.atum.agent.model.MeasureResult
+import za.co.absa.atum.model.Partitioning
+import za.co.absa.atum.model.dto.{AtumContextDTO, CheckpointDTO}
 
 trait Dispatcher {
-  def publish(checkpointKey: String, measureResult: MeasureResult): Unit
+  def fetchAtumContext(partitioning: Partitioning, parentPartitioning: Option[Partitioning]): Option[AtumContextDTO]
```

Comment:
```suggestion
  def getOrCreateAtumContext(partitioning: Partitioning, parentPartitioning: Option[Partitioning]): AtumContextDTO
```

server will always create if not exists and always return the object